lui.el

```diff
@@ -1333,5 +1390,135 @@ which see for an explanation of the argument BUFFER-STRING."
                            front-sticky t))))
 
 
+;;;;;;;;;;;;;;;;;;
+;;; Navigation ;;;
+;;;;;;;;;;;;;;;;;;
+
+(defun lui-at-message-p ()
+  "Check if point is on a message."
+  (get-text-property (point) 'lui-message-id))
+
+(defun lui-beginning-of-message-p ()
+  "Check if point is at the beginning of a message."
+  (not (equal (get-text-property (point) 'lui-message-id)
+              (get-text-property (1- (point)) 'lui-message-id))))
+
```

Comment:
Indeed it doesn't. I'm adding `(or (= (point) (point-min)) ...)` around `not`, since the beginning of a buffer has to be the beginning of a message. Even if there could be only half a message, the beginning of the buffer should still be considered the beginning of that message. At least I think so.